## Milestone 3

Goals: 
- Continue fixing, pre-processing, and preparing the data for your project
- Put it into DataFrame
- Visualize data
- Modeling 

I want to test my model on the following phrases:
- down at the base of a dizzy world
- the blink of a mountain
- the breadth of blood
- the crevassses of my present moment

My training dataset will be all of the writing. I have finished compiling it. See the txt file below. 

In [2]:
updatedpoems = open("./poetrydatasetml.txt", encoding='utf8').read()

# how data frame should look 
# title, list of lines, list of words

poemdictionary = {
    'Title': [],
    'Lines': [],
    'Words': []
}

In [3]:
updatedpoems

"\n\nCycles\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\nWritten and Illustrated by Miya Sheker\n\n\n\n\nfor Kristine\nTable of Contents\n\nIntroduction\t6\n\nWhenever it’s convenient!\t8\nLarger than Life\t9\nGoogle Keep\t10\nGoogle Keep 2\t11\nMarco?\t12\nVision and Voice\t13\nAspiration\t14\nAdmiration\t16\nOctober Thoughts\t17\nRecycle Bin\t18\nThere is something here. I promise.\t20\nWriter’s Block\t21\nOceans\t23\nBones, bones, bones\t24\nEveryone is Drowning\t26\nLetting go is the right thing.\t28\nPandora\t29\nAnswers Pending\t31\nPrescription\t32\nUniverse (circa 2018)\t34\nUniverse Reclaimed\t35\nUniverse Reimagined\t36\nOops\t37\nMatchstick\t39\nSleepwalker\t41\nIt’s a jungle out here.\t43\nRumbles\t45\nCrows and flowers\t47\nPebbles and Rocks\t48\nLava\t49\nMy Melatonin Collects Dust\t51\nPassersby\t53\nIn search of silhouettes\t55\nCeramic\t56\nWelcome home, honey.\t58\nTiptoe\t59\nMeet me at the playground\t61\nDemigod\t62\nIt’s okay to say goodbye.\t64\nHermit Crab\t65\nLolo Ano

I should go through the dataset and be able to grab "title" if there are two lines before (\n\n) and grab a line for everything that is followed by \n (and may need to remove the titles and the extra \n) and then grab words by splitting by " ". And then form the dataframe accordingly. 

Now, I will go and make sure that the poems in poetrydatasetml have consistent formatting so it will be easier to load into a dataframe. 

In [3]:
lines = updatedpoems.split("\n\n")
lines

['',
 'Introduction',
 'Cycles can be stupid. You know what I mean?',
 'Hi reader,',
 'This is my first letter to you. Thank you for picking up Cycles. I both hope, for your own sake, that you do not relate to the bulk of its contents and, for my own selfish need to find empathy in others, that you do see yourself in a poem or two. If you don’t know me, hello. It’s nice to meet you. My name is Miya. I am scared of crossing the street when the light is red, even without a car in sight. Sometimes, I set out on walks only to stop at random, just to see what I can find. My soul gravitates to large cities; I find peace in the bustle. I believe good art can change the trajectory of your day, maybe even your life. Cheesy? Yes. True? Also yes. At least for me. ',
 'In 2022, I published a little poetry book, Full, with my dear friend, Aurelia. Together, we explored the breadth of the human experience: the desire to create, to connect, to comprehend the world that we are living in. I found mysel

In [4]:
titles = []
nontitles = []

for l in lines: 
    if l.startswith("\n") == True:
        titles.append(l)
    else:
        nontitles.append(l)

titles = [l[1:] for l in titles]

In [5]:
titles

['Whenever it’s convenient!',
 'Larger than Life ',
 'Google Keep',
 'Google Keep 2',
 'Marco?',
 'Vision and Voice',
 'Aspiration',
 'Handwritten',
 'Admiration',
 'October Thoughts',
 'Recycle Bin',
 'There is something here. I promise.',
 'Writer’s Block',
 'Handwritten',
 'Oceans',
 'Bones, bones, bones',
 'Handwritten',
 'Everyone is Drowning',
 'Letter to reader',
 'Letting go is the right thing.',
 'Pandora  ',
 'Answers Pending',
 'Prescription',
 'Handwritten',
 'Universe (circa 2018)',
 'Universe Reclaimed',
 'Universe Reimagined',
 'Oops',
 'This is not about heartbreak',
 'Matchstick',
 'Sleepwalker',
 'Handwritten',
 'Handwritten',
 'It’s a jungle out here.',
 'Letter',
 'Rumbles ',
 'Crows and flowers',
 'Handwritten',
 'Pebbles and Rocks',
 'Lava',
 'Handwritten',
 'My Melatonin Collects Dust',
 'Passersby',
 'Handwritten',
 'Handwritten ',
 'In search of silhouettes',
 'Ceramic ',
 'Handwritten',
 'Welcome home, honey.',
 'Tiptoe ',
 'Handwritten',
 'Meet me at the play

In [6]:
import matplotlib.pyplot as plt
import pandas as pd

from sklearn.preprocessing import OrdinalEncoder


In [27]:
nontitles

['',
 'Introduction',
 'Cycles can be stupid. You know what I mean?',
 'Hi reader,',
 'This is my first letter to you. Thank you for picking up Cycles. I both hope, for your own sake, that you do not relate to the bulk of its contents and, for my own selfish need to find empathy in others, that you do see yourself in a poem or two. If you don’t know me, hello. It’s nice to meet you. My name is Miya. I am scared of crossing the street when the light is red, even without a car in sight. Sometimes, I set out on walks only to stop at random, just to see what I can find. My soul gravitates to large cities; I find peace in the bustle. I believe good art can change the trajectory of your day, maybe even your life. Cheesy? Yes. True? Also yes. At least for me. ',
 'In 2022, I published a little poetry book, Full, with my dear friend, Aurelia. Together, we explored the breadth of the human experience: the desire to create, to connect, to comprehend the world that we are living in. I found mysel

Looking at nontitles, this would be tough to parse through since the poems are different line lengths. Let's go back to lines and see if there is a way to parse through between lines that are present in titles. Thinking through this, I think it would be better to try again by separating the poems first. Based on what came out of the others, it looks like each poem is separated by 3 line breaks, not 2.

In [11]:
test = updatedpoems.split("\n\n\n")
test[0]

'\n\nIntroduction\n\nCycles can be stupid. You know what I mean?\n\nHi reader,\n\nThis is my first letter to you. Thank you for picking up Cycles. I both hope, for your own sake, that you do not relate to the bulk of its contents and, for my own selfish need to find empathy in others, that you do see yourself in a poem or two. If you don’t know me, hello. It’s nice to meet you. My name is Miya. I am scared of crossing the street when the light is red, even without a car in sight. Sometimes, I set out on walks only to stop at random, just to see what I can find. My soul gravitates to large cities; I find peace in the bustle. I believe good art can change the trajectory of your day, maybe even your life. Cheesy? Yes. True? Also yes. At least for me. \n\nIn 2022, I published a little poetry book, Full, with my dear friend, Aurelia. Together, we explored the breadth of the human experience: the desire to create, to connect, to comprehend the world that we are living in. I found myself on a

In [12]:
test_titles = []
test_lines = []

for t in test:
    a = t.split("\n\n")
    print(a[:3])
    test_titles.append(a[0])
    test_lines.append(a[1:])

['', 'Introduction', 'Cycles can be stupid. You know what I mean?']
['Whenever it’s convenient!', 'Screaming\nI want you to see me\nPlease could you just fucking look', 'Pleading\nI made it so easy\nI wrote it all in this fucking book']
['Larger than Life ', 'Biding my time,\nI am in no rush.\nPaintbrush bristle fingertips\ngentle, circle\nalong my skin, dip\nin my aura, mix \nin the color.', 'Bending my timelines,\nthe longer I bathe in \nthe memories, the most potent \nfloat to the surface, materializing \nin pigment, watercolor washes \nfrom the fervor I work in.\nAn outpouring on a spatial plane,\na colored canvas, a place\nto be.']
['Google Keep', 'poems that sit in notes on my phone\nlittered vulnerabilities\namongst flowers of life\nflurries of petals encircle me gently\nuntil a vortex appears\na pitfall out of my present moment\na hyper-fixation on making sense\nof nothing and everything\ni am still searching for color\nits symbolism in paintings of the present\nas if capturing

In [14]:
test_titles[0] = "Introduction"

In [15]:
test_titles

['Introduction',
 'Whenever it’s convenient!',
 'Larger than Life ',
 'Google Keep',
 'Google Keep 2',
 'Marco?',
 'Vision and Voice',
 'Aspiration',
 'Handwritten',
 'Admiration',
 'October Thoughts',
 'Recycle Bin',
 'There is something here. I promise.',
 'Writer’s Block',
 'Handwritten',
 'Oceans',
 'Bones, bones, bones',
 'Handwritten',
 'Everyone is Drowning',
 'Letter to reader',
 'Letting go is the right thing.',
 'Pandora  ',
 'Answers Pending',
 'Prescription',
 'Handwritten',
 'Universe (circa 2018)',
 'Universe Reclaimed',
 'Universe Reimagined',
 'Oops',
 'This is not about heartbreak',
 'Matchstick',
 'Sleepwalker',
 'Handwritten',
 'Handwritten',
 'It’s a jungle out here.',
 'Letter',
 'Rumbles ',
 'Crows and flowers',
 'Handwritten',
 'Pebbles and Rocks',
 'Lava',
 'Handwritten',
 'My Melatonin Collects Dust',
 'Passersby',
 'Handwritten',
 'Handwritten ',
 'In search of silhouettes',
 'Ceramic ',
 'Handwritten',
 'Welcome home, honey.',
 'Tiptoe ',
 'Handwritten',
 'Me

In [16]:
test_lines

[['Introduction',
  'Cycles can be stupid. You know what I mean?',
  'Hi reader,',
  'This is my first letter to you. Thank you for picking up Cycles. I both hope, for your own sake, that you do not relate to the bulk of its contents and, for my own selfish need to find empathy in others, that you do see yourself in a poem or two. If you don’t know me, hello. It’s nice to meet you. My name is Miya. I am scared of crossing the street when the light is red, even without a car in sight. Sometimes, I set out on walks only to stop at random, just to see what I can find. My soul gravitates to large cities; I find peace in the bustle. I believe good art can change the trajectory of your day, maybe even your life. Cheesy? Yes. True? Also yes. At least for me. ',
  'In 2022, I published a little poetry book, Full, with my dear friend, Aurelia. Together, we explored the breadth of the human experience: the desire to create, to connect, to comprehend the world that we are living in. I found mysel

In [24]:
words = []

for w in test_lines:
    a = " ".join(w)
    b = a.replace("\n", " ")
    c = b.split(" ")
    words.append(c)

In [25]:
words[1:3]

[['Screaming',
  'I',
  'want',
  'you',
  'to',
  'see',
  'me',
  'Please',
  'could',
  'you',
  'just',
  'fucking',
  'look',
  'Pleading',
  'I',
  'made',
  'it',
  'so',
  'easy',
  'I',
  'wrote',
  'it',
  'all',
  'in',
  'this',
  'fucking',
  'book'],
 ['Biding',
  'my',
  'time,',
  'I',
  'am',
  'in',
  'no',
  'rush.',
  'Paintbrush',
  'bristle',
  'fingertips',
  'gentle,',
  'circle',
  'along',
  'my',
  'skin,',
  'dip',
  'in',
  'my',
  'aura,',
  'mix',
  '',
  'in',
  'the',
  'color.',
  'Bending',
  'my',
  'timelines,',
  'the',
  'longer',
  'I',
  'bathe',
  'in',
  '',
  'the',
  'memories,',
  'the',
  'most',
  'potent',
  '',
  'float',
  'to',
  'the',
  'surface,',
  'materializing',
  '',
  'in',
  'pigment,',
  'watercolor',
  'washes',
  '',
  'from',
  'the',
  'fervor',
  'I',
  'work',
  'in.',
  'An',
  'outpouring',
  'on',
  'a',
  'spatial',
  'plane,',
  'a',
  'colored',
  'canvas,',
  'a',
  'place',
  'to',
  'be.',
  'Surrounded',
  '

In [26]:
print(len(test_titles))
print(len(test_lines))
print(len(words))

164
164
164


Yay!! It worked!!!

In [57]:
poems_df = pd.DataFrame(test_titles)

In [58]:
poems_df["Lines"] = test_lines
poems_df["Words"] = words

In [59]:
poems_df.columns = ["Title", "Lines", "Words"]

In [60]:
poems_df

Title  \
0                    Introduction   
1       Whenever it’s convenient!   
2               Larger than Life    
3                     Google Keep   
4                   Google Keep 2   
..                            ...   
159                     untitled    
160                         notes   
161                  quadrant one   
162                  quadrant two   
163  hypertext translation portal   

                                                 Lines  \
0    [Introduction, Cycles can be stupid. You know ...   
1    [Screaming\nI want you to see me\nPlease could...   
2    [Biding my time,\nI am in no rush.\nPaintbrush...   
3    [poems that sit in notes on my phone\nlittered...   
4    [poems that sit in notes on my phone\nscribble...   
..                                                 ...   
159  [this pain is a reminder of your capacity to l...   
160  [i\n have this gnawing fear of the end\n not t...   
161  [i have always sought a mirror\na portal, a pl...   
162  [admiration hand in hand with\nhesitation\nso ...   
163  [You think I’d be angry but I am not. \nI made...   

                                                 Words  
0    [Introduction, Cycles, can, be, stupid., You, ...  
1    [Screaming, I, want, you, to, see, me, Please,...  
2    [Biding, my, time,, I, am, in, no, rush., Pain...  
3    [poems, that, sit, in, notes, on, my, phone, l...  
4    [poems, that, sit, in, notes, on, my, phone, s...  
..                                                 ...  
159  [this, pain, is, a, reminder, of, your, capaci...  
160  [i, , have, this, gnawing, fear, of, the, end,...  
161  [i, have, always, sought, a, mirror, a, portal...  
162  [admiration, hand, in, hand, with, hesitation,...  
163  [You, think, I’d, be, angry, but, I, am, not.,...  

[164 rows x 3 columns]

I will save visualization for the next milestone; let's look at some stats for each poem by adding in a length.

In [64]:
length = []

for w in words:
    length.append(len(w))

length[:3]
print(len(length))

164


In [63]:
poems_df["Length"] = length

In [65]:
poems_df.head()

Title  \
0               Introduction   
1  Whenever it’s convenient!   
2          Larger than Life    
3                Google Keep   
4              Google Keep 2   

                                               Lines  \
0  [Introduction, Cycles can be stupid. You know ...   
1  [Screaming\nI want you to see me\nPlease could...   
2  [Biding my time,\nI am in no rush.\nPaintbrush...   
3  [poems that sit in notes on my phone\nlittered...   
4  [poems that sit in notes on my phone\nscribble...   

                                               Words  Length  
0  [Introduction, Cycles, can, be, stupid., You, ...     550  
1  [Screaming, I, want, you, to, see, me, Please,...      27  
2  [Biding, my, time,, I, am, in, no, rush., Pain...     180  
3  [poems, that, sit, in, notes, on, my, phone, l...      72  
4  [poems, that, sit, in, notes, on, my, phone, s...      92

In [66]:
print(poems_df["Length"].min())
print(poems_df["Length"].max())
print(poems_df["Length"].mean())
print(poems_df["Length"].median())
print(poems_df["Length"].std())




3
747
95.08536585365853
65.0
101.89170254898869


As for stats, generally my poems are 95 words long on average, with the max being 747. The large max makes sense as introductions to books and other prose pieces are included in the dataset.